python/ray/serve/_private/autoscaling_policy.py

```diff
@@ -202,3 +221,46 @@ def get_decision_num_replicas(
             self.decision_counter = 0
 
         return decision_num_replicas
+
+    def set_target_capacity(
+        self,
+        new_target_capacity: Optional[float],
+        new_target_capacity_scale_direction: Optional[TargetCapacityScaleDirection],
+    ):
+        self.target_capacity = new_target_capacity
+        self.scale_direction = new_target_capacity_scale_direction
+
+    def get_capacity_adjusted_min_replicas(self) -> int:
+        return get_capacity_adjusted_num_replicas(
+            self.config.min_replicas, self.target_capacity
+        )
+
+    def get_capacity_adjusted_initial_replicas(self) -> Optional[int]:
+        if self.config.initial_replicas is None:
+            return None
+        else:
+            return get_capacity_adjusted_num_replicas(
+                self.config.initial_replicas, self.target_capacity
+            )
+
+    def get_capacity_adjusted_max_replicas(self) -> int:
+        return get_capacity_adjusted_num_replicas(
+            self.config.max_replicas, self.target_capacity
+        )
+
+    def apply_bounds(self, curr_target_num_replicas: int) -> int:
+        """Gets a valid target_num_replicas using the current value and bounds."""
+
+        curr_upper_bound = self.get_capacity_adjusted_max_replicas()
+        curr_lower_bound = self.get_curr_lower_bound()
+
+        return max(curr_lower_bound, min(curr_upper_bound, curr_target_num_replicas))
+
+    def get_curr_lower_bound(self) -> int:
+        if self.get_capacity_adjusted_initial_replicas() is not None and (
+            self.scale_direction is None
+            or self.scale_direction == TargetCapacityScaleDirection.UP
+        ):
+            return self.get_capacity_adjusted_initial_replicas()
```

Comment:
Why does initial replicas factor into the lower bound during autoscaling? Shouldn't this just be set when the deployment is deployed/updated?

Reply:
While the `target_capacity` is scaling up, we scale `initial_replicas` by `target_capacity` and use that as a lower bound. This makes sense semantically because `initial_replicas` should be the number of replicas that a deployment starts with, and a deployment hasn't quite started until `target_capacity` has finished scaling up.